backend/inspirehep/records/api/base.py

```diff
@@ -159,6 +188,9 @@ def create(cls, data, id_=None, *args, **kwargs):
             kwargs.pop("disable_relations_update", None)
             data["self"] = get_ref_from_pid(cls.pid_type, data["control_number"])
             record = super().create(data, id_=id_, **kwargs)
+
+            if not record.get("deleted") and record.get("deleted_records"):
```

Comment:
are you sure the first clause is needed? what happens if you set a record to `deleted` and simultaneously add `deleted_records`?

Reply:
Question is do we want to redirect to deleted record? Currently this wont' be possible. You can redirect to a record then delete it, but you cannot redirect to deleted record or to record you are just deleting or to a new record you are creating in deleted state already.